pkg/server/server.go

```diff
@@ -230,3 +232,54 @@ func GetClaimsHandler(service types.Querier) http.HandlerFunc {
 		}
 	}
 }
+
+// Document is a did document that describes a did subject.
+// See https://www.w3.org/TR/did-core/#dfn-did-documents.
+type Document struct {
+	Context            []string             `json:"@context"` // https://w3id.org/did/v1
+	ID                 string               `json:"id"`
+	Controller         []string             `json:"controller,omitempty"`
+	VerificationMethod []VerificationMethod `json:"verificationMethod,omitempty"`
+	Authentication     []string             `json:"authentication,omitempty"`
+	AssertionMethod    []string             `json:"assertionMethod,omitempty"`
+}
+
+// VerificationMethod describes how to authenticate or authorize interactions
+// with a did subject.
+// See https://www.w3.org/TR/did-core/#dfn-verification-method.
+type VerificationMethod struct {
+	ID                 string `json:"id,omitempty"`
+	Type               string `json:"type,omitempty"`
+	Controller         string `json:"controller,omitempty"`
+	PublicKeyMultibase string `json:"publicKeyMultibase,omitempty"`
+}
+
+// GetDIDDocument returns the DID document for did:web resolution.
+func GetDIDDocument(id principal.Signer) http.HandlerFunc {
+	doc := Document{
+		Context: []string{"https://w3id.org/did/v1"},
+		ID:      id.DID().String(),
+	}
+	if s, ok := id.(signer.WrappedSigner); ok {
+		vid := fmt.Sprintf("%s#owner", id.DID())
+		doc.VerificationMethod = []VerificationMethod{
+			{
+				ID:                 vid,
+				Type:               "Ed25519VerificationKey2020",
+				Controller:         id.DID().String(),
+				PublicKeyMultibase: strings.TrimPrefix(s.Unwrap().DID().String(), "did:key:"),
+			},
+		}
+		doc.Authentication = []string{vid}
+		doc.AssertionMethod = []string{vid}
+	}
```

Comment:
nit: maybe always use `s` rather than `id` in the `if`'s body since it has already been type-asserted?
```suggestion
	if s, ok := id.(signer.WrappedSigner); ok {
		vid := fmt.Sprintf("%s#owner", s.DID())
		doc.VerificationMethod = []VerificationMethod{
			{
				ID:                 vid,
				Type:               "Ed25519VerificationKey2020",
				Controller:         s.DID().String(),
				PublicKeyMultibase: strings.TrimPrefix(s.Unwrap().DID().String(), "did:key:"),
			},
		}
		doc.Authentication = []string{vid}
		doc.AssertionMethod = []string{vid}
	}
```